### PR TITLE
autotest.parallel_dd: define target disk

### PR DESCRIPTION
**Pull request by YongxueHong**
Define a target disk with the parameter serial

Signed-off-by: Yongxue Hong <yhong@redhat.com>

ID: 1474001

### CODE REVIEW

**Comment by lmr** [comment]
Please rebase against the latest master, this should fix the Travis CI job for you.

**Comment by YongxueHong** [comment]
PASS 1-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.ppc64le.io-github-autotest-qemu.autotest.parallel_dd
PASS 2-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_scsi.up.virtio_net.RHEL.7.4.ppc64le.io-github-autotest-qemu.autotest.parallel_dd

PASS 1-smp_8.8192m.repeat1.run_test.Host_RHEL.RHEL_ALT.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.ALT.7.4.ppc64le.io-github-autotest-qemu.autotest.parallel_dd
PASS 2-smp_8.8192m.repeat1.run_test.Host_RHEL.RHEL_ALT.m7.u4.qcow2.virtio_scsi.up.virtio_net.RHEL.ALT.7.4.ppc64le.io-github-autotest-qemu.autotest.parallel_dd

File: shared/control/parallel_dd.control [hunk 1]
```diff
@@ -17,8 +17,8 @@ path.sort()
 fs = path[-1]
 platform = Popen(["uname", "-m"], stdout=PIPE).communicate()[0]
 if "ppc" in platform:
-    if not Popen("lsscsi", stdout=PIPE).communicate()[0]:
-        fs = path[0]
+    path = glob.glob("/dev/disk/by-id/*TARGET_DISK*")
+    fs = ('/dev/' + Popen(["ls", path[0], "-ls"], stdout=PIPE).stdout.readline().split('/')[-1]).split('\n')[0]
```
Comment (Zhengtong) [inline shared/control/parallel_dd.control]: If the behavior of x86 and ppc is identical.  How about remove 'if "ppc" in platform' ?  And let the x86 and ppc go with the same logic ?

Comment (YongxueHong) [inline shared/control/parallel_dd.control]: ok, I will update it.

**Comment by Zhengtong** [comment]
LGTM.  Ack-by: Zhengtong <zhengtli@redhat.com>

File: shared/control/parallel_dd.control [hunk 2]
```diff
@@ -12,13 +12,8 @@ the files system.
...
 
 # YOU NEED TO SPECIFY A FILESYSTEM
-path = glob.glob("/dev/[hsv]d?")
-path.sort()
-fs = path[-1]
-platform = Popen(["uname", "-m"], stdout=PIPE).communicate()[0]
-if "ppc" in platform:
-    if not Popen("lsscsi", stdout=PIPE).communicate()[0]:
-        fs = path[0]
+path = glob.glob("/dev/disk/by-id/*TARGET_DISK*")
```
Comment (hereischen) [inline shared/control/parallel_dd.control]: It seems `glob` was not imported in this script.

Comment (hereischen) [inline shared/control/parallel_dd.control]:
```
>>> from subprocess import Popen, PIPE
>>> 
>>> path = glob.glob("/dev/disk/by-id/*qazwsx*")
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'glob' is not defined

Comment (YongxueHong) [inline shared/control/parallel_dd.control]: hi, the following is the part of function called:
workspace/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/generic/tests/cfg/autotest_control.cfg
	- parallel_dd:
		blk_extra_params_stg = "serial=TARGET_DISK0"
		test_timeout = 36000
		images += " stg"
		image_format_stg = qcow2
		boot_drive_stg = yes
		image_name_stg = images/storage
		image_size_stg = 10G
		force_create_image_stg = yes
		test_control_file = parallel_dd.control

workspace/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/generic/tests/autotest_control.py
def run(test, params, env):
	...
	control_path = os.path.join(test.virtdir, "control",params.get("test_control_file"))  ==> test_control_file = parallel_dd.control
	...
	utils_test.run_autotest(vm, session, control_path, timeout, outputdir,params, control_args=control_args,ignore_session_terminated=ignore_sess_terminated)
		def run_autotest(vm, session, control_path, timeout,outputdir, params, copy_only=False, control_args=None,ignore_session_terminated=False):
			...
			temp_control_path = config_control(control_path, job_args=job_args)
			vm.copy_files_to(temp_control_path, os.path.join(destination_autotest_path, 'control'))
			...
			session.cmd_output("python -x ./autotest-local %s control & wait ${!}" %verbose,timeout=timeout,print_func=logging.info)
			workspace/lib/python2.7/site-packages/autotest/client/autotest_local.py  ==> I think this file will be called
				...
				from autotest.client import job
					workspace/lib/python2.7/site-packages/autotest/client/job.py
					...
					import glob  ==> here import the glob module..
					...
				def main(self):
					job.runjob(os.path.realpath(self.args[0]), drop_caches, self.options)  ==>Run a job using the given control file.
						myjob = job(control=control, drop_caches=drop_caches, options=options)
						...

for autotest flow, please refer to : http://autotest.readthedocs.io/en/latest/main/local/index.html

**Comment by hereischen** [comment]
Ack-by: Haotong Chen hachen@redhat.com

**Comment by XueqiangWei** [comment]
LGTM,
Acked-by xuwei@redhat.com

File: shared/control/parallel_dd.control [hunk 3]
```diff
@@ -12,13 +12,8 @@ the files system.
...
+fs = ('/dev/' + Popen(["ls", path[0], "-ls"], stdout=PIPE).stdout.readline().split('/')[-1]).split('\n')[0]
```
Comment (aliang123) [inline shared/control/parallel_dd.control]: I think fs = ('/dev/' + Popen(["ls", path[0], "-ls"], stdout=PIPE).stdout.readline().split('/')[-1]).strip("\n") will be better.

Comment (YongxueHong) [reply shared/control/parallel_dd.control]: Good advice , I will update it , thank you very much.

**Comment by aliang123** [comment]
Acked-by: Aihua Liang<aliang@redhat.com>

Test cmd:
  python ConfigTest.py --guestname=RHEL.7.4 --verbose=yes --testcase=parallel_dd
Test result:
    PASS 1-smp_8.65536m.repeat1.run_test.Host_RHEL.RHEL_ALT.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.ppc64.io-github-autotest-qemu.autotest.parallel_dd
PASS 2-smp_8.65536m.repeat1.run_test.Host_RHEL.RHEL_ALT.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.ppc64le.io-github-autotest-qemu.autotest.parallel_dd

**Pull request closed without merging**